qgis_plugin_gpq_downloader.py

```diff
@@ -661,36 +662,54 @@ def run(self, default_source=None):
         # Connect validation complete signal to handle the result
         dialog.validation_complete.connect(
             lambda success, message, results: self.handle_validation_complete(
-                success, message, results, dialog.get_url(), self.iface.mapCanvas().extent()
+                success, message, results, dialog.get_url(), self.iface.mapCanvas().extent(), dialog
             )
         )
         
         if dialog.exec_() != QDialog.Accepted:
             return
 
-    def handle_validation_complete(self, success, message, validation_results, dataset_url, extent):
-        """Handle validation completion and start download if successful"""
+    def handle_validation_complete(self, success, message, validation_results, url, extent, dialog):
+        """Handle validation completion and start download if successful."""
         if success:
-            # Generate default filename
-            from datetime import datetime
-            timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
-            default_filename = f"geoparquet_download_{timestamp}.parquet"
-            default_save_path = str(self.download_dir / default_filename)
+            # Get current date for filename
+            current_date = datetime.datetime.now().strftime('%Y%m%d')
             
+            # Generate the default filename based on dialog selection
+            if dialog.overture_radio.isChecked():
+                theme = dialog.overture_combo.currentText().lower()
+                if theme == "base":
+                    subtype = dialog.base_subtype_combo.currentText()
+                    filename = f"overture_base_{subtype}_{current_date}.parquet"
+                else:
+                    filename = f"overture_{theme}_{current_date}.parquet"
+            
+            elif dialog.sourcecoop_radio.isChecked():
+                selection = dialog.sourcecoop_combo.currentText()
+                # Convert display name to safe filename format
+                safe_name = selection.lower().replace(" ", "_").replace("/", "_")
```

Comment:
One little observation here is that the vida name became quite long. But we could also just use this to nudge us to shorter names...